woocommerce/woocommerce
# Show notice for expired and expiring subscriptions in settings and in-app extensions page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

Show notice on the WC settings page and marketplace page about expired and expiring subscriptions


Notices we want to show:

- Expired subscriptions
- Expiring Subscriptions expiring (Subscription expiring soon)


### Changes proposed in this Pull Request:

- Added notice in WC core settings page and extensions page

Note about notices:
- Notice will show when the store are connected with woo commerce.com
- Those should be dismissable
- Once dismissed, they should not appear again for the user next time:

Messages we want to show:
- expired subscriptions
   - single expired subscription for product X and the user has only a single subscription to X: `Your subscription for [product name] expired. Renew for [price] to continue receiving updates and streamlined support.`
   - single expired subscription for product X and the user has multiple subscriptions to X: `One of your subscriptions for [product name] has expired. Renew for [price]  to continue receiving updates and streamlined support.`
   - multiple expired subscriptions: `You have X Woo extension subscriptions that expired. Renew to continue receiving updates and streamlined support.`
- subscriptions expiring soon
   - single subscription for product X is about to expire and the user has only a single subscription to X: `Your subscription for [product name] expires on [date]. Enable auto-renewal to continue receiving updates and streamlined support.` 
   - single expired subscription for product X and the user has multiple subscriptions to X: `One of your subscriptions for [product name] expires on [date]. Enable auto-renewal to continue receiving updates and streamlined support.`
   - multiple expired subscriptions: `:You have X Woo extension subscriptions expiring soon. Enable auto-renewal to continue receiving updates and streamlined support.`




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build WooCommerce on your local environment, e.g.: `cd plugins/woocommerce` then `nvm use && pnpm run build --filter=woocommerce`.
2. Connected your store with local woocommerce.com test site. 
4. Make sure you have no expired subscriptions and no subscriptions expiring in the next 7 days
5. Go to the WC setting page and Extension page and confirm you see no notices related to subscriptions
6. Make sure you have 2 paid subscriptions for the same product, note their order IDs: <order_id_a>, <order_id_b>
7. Set one of the orders as expiring soon by setting an expiry date in the next 7 days. 


![Screenshot 2024-04-30 at 6 47 39 PM](https://github.com/woocommerce/woocommerce/assets/10769293/feabf658-38b0-435e-a6b7-46c04e755204)

8. Set one of the orders as expiring soon by setting an expiry date in the next 7 days (From same two products). To do that, run [this query](https://github.com/Automattic/woocommerce.com/issues/20256#issuecomment-2111773279)

![Screenshot 2024-04-30 at 7 00 53 PM](https://github.com/woocommerce/woocommerce/assets/10769293/9dfb4c19-258a-47ba-8f29-283670dd0c1a)

9. Make sure that you have  different products expiring soon, you should see a different message now:

![Screenshot 2024-04-30 at 6 58 38 PM](https://github.com/woocommerce/woocommerce/assets/10769293/3bf9bfcf-f988-4fb5-bc6a-cc906b937435)

10. Set one of the subscriptions as expired 

![Screenshot 2024-04-30 at 1 56 20 AM](https://github.com/woocommerce/woocommerce/assets/10769293/53659893-32bf-45f6-8b16-3b217122b7c6)

11. Add more expired subscriptions and confirm that message changes properly
12. Dismiss the notice and refresh the page. You shouldn't see notice about expired/expiring subscriptions.

![Screenshot 2024-04-30 at 1 58 38 AM](https://github.com/woocommerce/woocommerce/assets/10769293/071c2578-f791-4707-b297-502327fed960)

![Screenshot 2024-04-30 at 2 02 57 AM](https://github.com/woocommerce/woocommerce/assets/10769293/b634a4f7-3020-4c20-9db7-81c6d9056873)




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
